docs/api/webob.txt

```diff
@@ -16,9 +16,25 @@ Accept-*
    :members:
    :inherited-members:
 
+.. autofunction:: create_accept_language_header
+
 .. autoclass:: AcceptLanguage
-   :members:
-   :inherited-members:
+   :members: parse
+
+.. autoclass:: AcceptLanguageValidHeader
+   :members: header_value, parsed, __init__, __add__, __contains__, __iter__,
+	     __radd__, __str__, parse, basic_filtering, best_match, lookup,
+             quality
+
+.. autoclass:: AcceptLanguageNoHeader
+   :members: header_value, parsed, __init__, __add__, __contains__, __iter__,
+	     __radd__, __str__, parse, basic_filtering, best_match, lookup,
+             quality
+
+.. autoclass:: AcceptLanguageInvalidHeader
+   :members: header_value, parsed, __init__, __add__, __contains__, __iter__,
+	     __radd__, __str__, parse, basic_filtering, best_match, lookup,
+             quality
```

Comment:
Can we remove the subclasses from the public api?

Reply:
We can if we combine the docstrings for all the subclasses for each method and put them all into `AcceptLanguage`. This would imo make some already complicated docstrings even more complicated, adding another layer to the branching. And as I mentioned before, the instances would still i.d. as the subclasses (I guess we could change the `__repr__` to indicate that it is a subclass of `AcceptLanguage`, so people would know to look there?) If you want me to go ahead and make the change, I will &mdash; it's not trivial, and will take some time and thought on how to combine the docstrings from the three subclasses for each method. (I would also have to do the same for the other three headers.)

Would it be possible to merge this so I can push changes for this and the other three headers before the deadline tomorrow to a more appropriately-named PR, or should I push the changes for all four headers to this one?

Reply:
It's fine if we want to merge this now. I'm happy with that.

Part of the idea of making the subclasses "hidden" is not that people don't know about them but that they don't care. I'm more worried about the docs the user sees than the actual objects they see in the repl. With that in mind we'd want to make the docstrings focus on valid headers with just a sentence or two in each that explains what happens with invalid/missing data. I can't imagine that will make the docstrings that much more complex or harder to parse but I could be wrong!

Reply:
I was worrying whether it would be clear enough that people should look at the docs for `AcceptLanguage` if the objects identify as the subclasses, but I understand what you are looking for, and will try to combine the docstrings and document one api under the base class after I've finished wrapping up the four headers and all the necessary submissions for gsoc. Could you please merge this PR if everything is ok, so I can open a new PR and push the other changes? (Sorry about the last-minute commits, they are minor changes to fix small mistakes.) Thanks!

Reply:
Thanks again @whiteroses I'm happy with all of this.

Reply:
Alright, any future work can be done in a different PR.